### DISCUSSION TITLE
Return custom HTTP Status Code with appDir server component

### DISCUSSION THREAD

**Discussion by ben519** [vercel/next.js#52796] [App Router]
This question has been asked several times for specific use-cases

- [Returning 500 or 503 Error Response Status Code from Layout or Page](https://github.com/vercel/next.js/discussions/52378)
- [How to return custom status code (429) with React Server Components?](https://github.com/vercel/next.js/discussions/50149)
- [Custom error code for notFound page in appDir](https://github.com/vercel/next.js/discussions/50906)
- [Set different unsuccessful status codes on page (RSC) when fetching or something else goes wrong?](https://github.com/vercel/next.js/discussions/50383)

## My specific use case:
In my case, when a user requests a page he doesn't have permission to view, I return a `<NotAuthorized />` component like so

```
// user/[userId]/edit/page.jsx (server component)

import { EditProfile } from '@/components/Profile'
import NotAuthorized from '@/components/NotAuthorized'

export default async function Page({ params }) {
  // setup code here..

  if (user.id !== params.userId) {
    return <NotAuthorized />
  }

  return (
    <div>
      <EditProfile userId={ params.userId } />
    </div>
  )
}
```

In the case where I return the `<NotAuthorized />` component, I would love to issue a [401 Unauthorized HTTP Status Code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401). 

## General Request
More generally, it'd be great if server components let me return *any* status code. Is this possible?

**Comment by devjiwonchoi** [comment]
Posted #53225 before working on it, can you confirm if this is what you are requesting?
